Ginger/Ginger/Actions/UserControls/UCFlowControlAction.xaml.cs

```diff
@@ -251,38 +288,68 @@ private void SetActionValueComboData()
                                     }
-                                    else if (FC.GetGuidFromValue(true) == Guid.Empty && FC.GetNameFromValue(true) == a.Description)
+
+                                    ComboEnumItem CEI = new ComboEnumItem();
+                                    CEI.Value = a.Guid + FC.GUID_NAME_SEPERATOR + a.ActivityName;//adding also name as second option search to be used when pulling the activity from Shared Repository
+                                    CEI.text = a.ActivityName;
+                                    ActionValueComboBox.Items.Add(CEI);
+
+                                    if (ActionValueComboBox.SelectedItem == null ||
+                                       (ActionValueComboBox.SelectedItem != null && a.Active))
                                     {
-                                        ActionValueComboBox.SelectedItem = CEI;
+                                        if (FC.GetGuidFromValue(true) == a.Guid)
+                                        {
+                                            ActionValueComboBox.SelectedItem = CEI;
+                                        }
+                                        else if (FC.GetGuidFromValue(true) == Guid.Empty && FC.GetNameFromValue(true) == a.Description)
+                                        {
+                                            ActionValueComboBox.SelectedItem = CEI;
+                                        }
+
                                     }
+                                }
+                                ActionValueComboBox.Visibility = System.Windows.Visibility.Visible;
+                                ActionValueTextBox.Visibility = System.Windows.Visibility.Hidden;
 
+                                if (FC.Value != null && ActionValueComboBox.SelectedItem == null)
+                                {
+                                    Reporter.ToUser(eUserMsgKey.ActivityIDNotFound, FC.Value);
                                 }
                             }
-                            ActionValueComboBox.Visibility = System.Windows.Visibility.Visible;
-                            ActionValueTextBox.Visibility = System.Windows.Visibility.Hidden;
-
-                            if (FC.Value != null && ActionValueComboBox.SelectedItem == null)
+                            else
                             {
-                                Reporter.ToUser(eUserMsgKey.ActivityIDNotFound, FC.Value);
+                                Activity activityLinkedToFC = null;
+                                if (FC.GetGuidFromValue(true) != null && FC.GetGuidFromValue(true) != Guid.Empty)
+                                {
+                                    activityLinkedToFC = WorkSpace.Instance.SolutionRepository.GetAllRepositoryItems<Activity>().Where(x => x.Guid == FC.GetGuidFromValue(true)).FirstOrDefault();
+                                }
+                                else if(activityLinkedToFC==null)
```

Comment:
this condition will not work when activity is not found by GUID.
Needs to improved to cover both scenarios